<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <xacro:macro name="desktop-arm_ros2_control" params="name initial_positions_file">
        <xacro:property name="initial_positions" value="${load_yaml(initial_positions_file)['initial_positions']}"/>

        <ros2_control name="${name}" type="system">
            <hardware>
                <plugin>fake_components/GenericSystem</plugin>
            </hardware>
            <joint name="joint_1">
                <param name="initial_position">${initial_positions['joint_1']}</param>
                <command_interface name="position" />
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>
            <joint name="joint_2">
                <param name="initial_position">${initial_positions['joint_2']}</param>
                <command_interface name="position" />
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>
            <joint name="joint_3">
                <param name="initial_position">${initial_positions['joint_3']}</param>
                <command_interface name="position" />
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>
            <joint name="joint_4">
                <param name="initial_position">${initial_positions['joint_4']}</param>
                <command_interface name="position" />
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>
            <joint name="joint_5">
                <param name="initial_position">${initial_positions['joint_5']}</param>
                <command_interface name="position" />
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>
            <joint name="joint_6">
                <param name="initial_position">${initial_positions['joint_6']}</param>
                <command_interface name="position" />
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>
        </ros2_control>

    </xacro:macro>

</robot>
